Testing

Mikearaya opened the pull request:
Hi
on this pull request, I have tested all the cases for the enumerable methods I've created
1. #my_each 
2. #my_each_with_index
3. #my_select (#my_find_all, #my_filter)
4. #my_all? 
5. #my_any?
6. #my_none?
7. #my_count 
8. #my_map

### I'm kindly requesting your review so I can proceed and merge all tests to develop and `master`

**Thank you** :bowing_man:

Mikearaya committed:
Add RSpec configuration

Mikearaya committed:
Add #my_all? test

Mikearaya committed:
Add my_any? tests

Mikearaya committed:
Add #my_count tests

Mikearaya committed:
Add #my_filter/#my_select tests

Mikearaya committed:
Add #my_find_all test

Mikearaya committed:
Add #my_inject and #my_reduce tests

Mikearaya committed:
Add #my_none? tests

Mikearaya committed:
Add #my_map  tests

Mikearaya committed:
Add #my_each_with_index tests

Mikearaya committed:
Add #my_each tests

mosesogwo commented on spec/iterator_spec.rb:
```diff
@@ -0,0 +1,220 @@
+  let(:number) { [1, 2, 3, 4, 2] }
+  let(:string) { %w[cat bear rat] }
+  context '#my_all?' do
+    it 'Return true when no block is given' do
+      expect(number.my_all?).to eq true
+    end
```
Include test for when no block is given and there is at least one element that is false or nil. In that case, it should return false

Mikearaya replied:
**Noted thanks**

mosesogwo commented on spec/iterator_spec.rb:
```diff
@@ -0,0 +1,220 @@
+  end
+
+  context '#my_any?' do
+    it 'Return true when no block is given' do
+      expect(number.my_any?).to eq true
+    end
```
When no block is given and none of the elements is a truthy, it should return false. Add test for that scenario

Mikearaya replied:
**Noted thanks**

mosesogwo commented on spec/iterator_spec.rb:
```diff
@@ -0,0 +1,220 @@
+
+  context '#my_filter should Return ' do
+    it 'items that match block expression' do
+      expect(number.my_filter(&:even?)).to all(be_even)
```
A better check will be to expect the exact array of even numbers. Any other set of even numbers will pass this test... but we don't just want it to return any set of even numbers.

Mikearaya replied:
**Noted thanks**

mosesogwo commented on spec/iterator_spec.rb:
```diff
@@ -0,0 +1,220 @@
+    end
+  end
+
+  context '#my_select should Return ' do
```
This is an alias for my_filter. One method test for all the aliases should be sufficient since the method definition is the same.

Mikearaya replied:
**Removed**

mosesogwo commented on spec/iterator_spec.rb:
```diff
@@ -0,0 +1,220 @@
+    end
+  end
+
+  context '#my_find_all should Return ' do
```
This is an alias for my_filter. One method test for all the aliases should be sufficient since the method definition is the same.

Mikearaya replied:
**Removed**

mosesogwo commented on spec/iterator_spec.rb:
```diff
@@ -0,0 +1,220 @@
+    end
+  end
+
+  context '#my_inject should Return ' do
```
inject and reduce are aliases, one test should be sufficient.

Mikearaya replied:
**removed**

mosesogwo reviewed (changes requested):
**Hi @Mikearaya, good job with your tests.:+1:. Please see my comments on your spec files**


_**Kindly make those changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

Mikearaya committed:
Add missing examples

Mikearaya committed:
Remove duplicate examples for alias methods

Mikearaya commented:
Hi, @mosesogwo Thank you for your thorough review and recommendations. 

I have added the missing examples for `#my_any?` and `#my_all?` methods. I have also removed duplicate tests made for alias methods (`#my_inject`, `#my_find_all` & `#my_select`).

### Now I'm kindly requesting another review so I can merge to develop id all checks out.

**Thank you** :bowing_man:

maelfosso reviewed (approved):
# Congratulations :clap: 

It's OK :smile: You can merge it now :+1: 
Good continuation :handshake: 

Mael FOSSO
